# Importation des bibliothèques

In [137]:
import statsmodels.api as sm
import pandas as pd
import math

# Chargement et Nettoyage du questionnaire

**Nota** : Avant leur import, les résultats du questionnaire ont été nettoyées des quelques données personnelles insérées dans les commentaires.

In [155]:
df = pd.read_csv("V1.1_recrutement_et_IA_sauf noms.csv", header=[1], sep=";")

## Contrôle visuel du questionnaire

In [156]:
df.head(10)

Colonnes       SID Heure de soumission Heure de complétion  \
0         3  25290150  11/01/2025 - 20:28  11/01/2025 - 20:28   
1         4  25290276  11/01/2025 - 20:51  11/01/2025 - 20:51   
2         5  25290441  11/01/2025 - 21:18  11/01/2025 - 21:18   
3         6  25291335  12/01/2025 - 03:53  12/01/2025 - 03:53   
4         7  25291387  12/01/2025 - 06:52  12/01/2025 - 06:52   
5         8  25291456  12/01/2025 - 07:52  12/01/2025 - 07:52   
6         9  25291533  12/01/2025 - 08:27  12/01/2025 - 08:27   
7        10  25291545  12/01/2025 - 08:30  12/01/2025 - 08:30   
8        11  25291576  12/01/2025 - 08:37  12/01/2025 - 08:37   
9        12  25291608  12/01/2025 - 08:47  12/01/2025 - 08:47   

  Heure de modification  Brouillon Adresse IP  UID Nom d'utilisateur  \
0    11/01/2025 - 20:28          0    0.0.0.0    0         Anonymous   
1    11/01/2025 - 20:51          0    0.0.0.0    0         Anonymous   
2    11/01/2025 - 21:18          0    0.0.0.0    0         Anonymous   
3    12/01/2025 - 03:53          0    0.0.0.0    0         Anonymous   
4    12/01/2025 - 06:52          0    0.0.0.0    0         Anonymous   
5    12/01/2025 - 07:52          0    0.0.0.0    0         Anonymous   
6    12/01/2025 - 08:27          0    0.0.0.0    0         Anonymous   
7    12/01/2025 - 08:30          0    0.0.0.0    0         Anonymous   
8    12/01/2025 - 08:37          0    0.0.0.0    0         Anonymous   
9    12/01/2025 - 08:47          0    0.0.0.0    0         Anonymous   

  Rédiger des CV ou des lettres de motivation  ...  \
0                               Plutôt opposé  ...   
1                           Totalement opposé  ...   
2                             Assez favorable  ...   
3                       Tout à fait favorable  ...   
4                     Ni favorable, ni opposé  ...   
5                             Assez favorable  ...   
6                             Assez favorable  ...   
7                       Tout à fait favorable  ...   
8                           Totalement opposé  ...   
9                             Assez favorable  ...   

  Lors de la négociation sur votre rémunération et autres avantages  \
0                          Ni réticent, ni favorable                  
1                                      Très réticent                  
2                                           Réticent                  
3                                      Très réticent                  
4                                           Réticent                  
5                                           Réticent                  
6                                      Très réticent                  
7                                      Très réticent                  
8                                      Très réticent                  
9                                      Très réticent                  

  Lors de la rédaction de votre contrat de travail  \
0                                        Favorable   
1                                         Réticent   
2                                        Favorable   
3                        Ni réticent, ni favorable   
4                        Ni réticent, ni favorable   
5                        Ni réticent, ni favorable   
6                                    Très réticent   
7                                    Très réticent   
8                                    Très réticent   
9                                        Favorable   

  Le respect de votre vie privée L'empreinte carbone de l'IA utilisée  \
0         C'est plutôt important        C'est important ou secondaire   
1         C'est plutôt important                                  NaN   
2           C'est très important              C'est plutôt secondaire   
3         C'est plutôt important              C'est plutôt secondaire   
4          C'est très secondaire                                  NaN   
5           C'est très important                 C'est très important   
6         C

## Suppression des colonnes à caractère technique sans données liées aux questions

In [157]:
df = df.drop(df.columns[0:9], axis=1)

In [158]:
df.head(2)

Rédiger des CV ou des lettres de motivation Améliorer votre profil LinkedIn  \
0                               Plutôt opposé                 Assez favorable   
1                           Totalement opposé                   Plutôt opposé   

  Préparer des entretiens d'embauche  \
0                  Totalement opposé   
1                      Plutôt opposé   

  Vous renseigner sur votre futur employeur  \
0                             Plutôt opposé   
1                   Ni favorable, ni opposé   

  Répondre plus rapidement à une offre  \
0                      Assez favorable   
1              Ni favorable, ni opposé   

  ChatGPT et autres IA génératives (génération de texte ou d'images)  \
0                                           à l'aise                   
1                                    Plutôt à l'aise                   

  Moteur de recherche (Chrome, Safari, Firefox)  \
0                               Plutôt à l'aise   
1                                      à l'aise   

  Application de navigation disponibles sur votre mobile ou dans votre voiture  \
0                                    Plutôt à l'aise                             
1                                    Plutôt à l'aise                             

  Autres usages professionnels ou scientifiques  \
0                               Plutôt à l'aise   
1                               Plutôt à l'aise   

  Pour sélectionner les meilleur•e•s candidat•e•s  ...  \
0                                     Très faible  ...   
1                             Ni faible, ni élevé  ...   

  Lors de la négociation sur votre rémunération et autres avantages  \
0                          Ni réticent, ni favorable                  
1                                      Très réticent                  

  Lors de la rédaction de votre contrat de travail  \
0                                        Favorable   
1                                         Réticent   

  Le respect de votre vie privée L'empreinte carbone de l'IA utilisée  \
0         C'est plutôt important        C'est important ou secondaire   
1         C'est plutôt important                                  NaN   

  L'absence de toute discrimination L'équité entre tous les candidats  \
0              C'est très important              C'est très important   
1              C'est très important              C'est très important   

      Le respect de la diversité  \
0  C'est important ou secondaire   
1           C'est très important   

  Aimeriez-vous savoir si votre recruteur a prévu d'utiliser l'IA dans le processus de recrutement ?  \
0                                    C'est important                                                   
1                                    C'est essentiel                                                   

  Quelle est votre tranche d'âge?  \
0                       25-34 ans   
1                       45-54 ans   

  Donnez votre avis sur une entreprise qui souhaiterait confier ses recrutements à une IA :   
0                                                NaN                                          
1                                                NaN                                          

[2 rows x 27 columns]

## Simplification du nommage des colonnes du questionnaire

In [159]:
# rename columns dealing with usefulness
df = df.rename(columns={"Rédiger des CV ou des lettres de motivation": "useful_resume_CV",
                  "Améliorer votre profil LinkedIn": "useful_linkedin",
                  "Préparer des entretiens d'embauche": "useful_interview",
                  "Vous renseigner sur votre futur employeur": "useful_intelligence",
                  "Répondre plus rapidement à une offre": "useful_speedy_response"
                  })

In [160]:
# rename columns dealing with easiness of use
df = df.rename(columns={"ChatGPT et autres IA génératives (génération de texte ou d'images)": "easiness_generative_AI",
                  "Moteur de recherche (Chrome, Safari, Firefox)": "easiness_search_engine",
                  "Application de navigation disponibles sur votre mobile ou dans votre voiture": "easiness_navigation",
                  "Autres usages professionnels ou scientifiques": "easiness_business_sciences"
                  })

In [161]:
# rename columns dealing with trust
df = df.rename(columns={"Pour sélectionner les meilleur•e•s candidat•e•s": "trust_best_applicants",
                   "Pour réduire le risque de discrimination" : "trust_discrimination",
                    "Pour un recrutement plus éthique": "trust_ethics",
                    "Pour diversifier les recrutements": "trust_diversity"
                  })

In [162]:
# rename columns dealing with acceptance
df = df.rename(columns={"Lors du tri initial des candidatures": "acceptance_sorting_resumes",
                        "Lors des tests de sélection": "acceptance_selection_tests",
                        "Lors des entretiens individuels": "acceptance_interviews",
                        "Lors de la discussion sur le périmètre de votre poste": "acceptance_negotiation",
                        "Lors de la négociation sur votre rémunération et autres avantages": "acceptance_wages_package",
                        "Lors de la rédaction de votre contrat de travail": "acceptance_contract"
                       })

In [163]:
# Rename columns dealing with ethics
df = df.rename(columns={"Le respect de votre vie privée": "ethics_privacy",
                        "L\'empreinte carbone de l\'IA utilisée": "ethics_carbon_footprint",
                        "L\'absence de toute discrimination": "ethics_no_discrimination",
                        "L\'équité entre tous les candidats": "ethics_fairness_for_applicants",
                        "Le respect de la diversité": "ethics_diversity"
                        })

In [164]:
# Rename separate questions
df = df.rename(columns={
    "Aimeriez-vous savoir si votre recruteur a prévu d\'utiliser l\'IA dans le processus de recrutement ?": "misc_recruiter_uses_AI",
    "Quelle est votre tranche d\'âge?": "misc_age_category",
    "Donnez votre avis sur une entreprise qui souhaiterait confier ses recrutements à une IA : ": "misc_free_text"
    })

In [165]:
df.head(10)

useful_resume_CV        useful_linkedin         useful_interview  \
0            Plutôt opposé        Assez favorable        Totalement opposé   
1        Totalement opposé          Plutôt opposé            Plutôt opposé   
2          Assez favorable  Tout à fait favorable    Tout à fait favorable   
3    Tout à fait favorable  Tout à fait favorable    Tout à fait favorable   
4  Ni favorable, ni opposé          Plutôt opposé          Assez favorable   
5          Assez favorable        Assez favorable          Assez favorable   
6          Assez favorable        Assez favorable  Ni favorable, ni opposé   
7    Tout à fait favorable  Tout à fait favorable    Tout à fait favorable   
8        Totalement opposé      Totalement opposé        Totalement opposé   
9          Assez favorable  Tout à fait favorable  Ni favorable, ni opposé   

       useful_intelligence   useful_speedy_response  \
0            Plutôt opposé          Assez favorable   
1  Ni favorable, ni opposé  Ni favorable, ni opposé   
2            Plutôt opposé    Tout à fait favorable   
3    Tout à fait favorable    Tout à fait favorable   
4  Ni favorable, ni opposé  Ni favorable, ni opposé   
5    Tout à fait favorable          Assez favorable   
6          Assez favorable          Assez favorable   
7    Tout à fait favorable    Tout à fait favorable   
8          Assez favorable        Totalement opposé   
9    Tout à fait favorable          Assez favorable   

       easiness_generative_AI easiness_search_engine  \
0                    à l'aise        Plutôt à l'aise   
1             Plutôt à l'aise               à l'aise   
2             Plutôt à l'aise        Plutôt à l'aise   
3             Plutôt à l'aise               à l'aise   
4        Quelques difficultés          En difficulté   
5  Ni aisance, ni difficultés               à l'aise   
6             Plutôt à l'aise               à l'aise   
7                    à l'aise               à l'aise   
8             Plutôt à l'aise        Plutôt à l'aise   
9  Ni aisance, ni difficultés        Plutôt à l'aise   

          easiness_navigation  easiness_business_sciences  \
0             Plutôt à l'aise             Plutôt à l'aise   
1             Plutôt à l'aise             Plutôt à l'aise   
2  Ni aisance, ni difficultés  Ni aisance, ni difficultés   
3                    à l'aise                    à l'aise   
4               En difficulté        Quelques difficultés   
5                    à l'aise                    à l'aise   
6                    à l'aise             Plutôt à l'aise   
7                    à l'aise                    à l'aise   
8             Plutôt à l'aise             Plutôt à l'aise   
9             Plutôt à l'aise             Plutôt à l'aise   

  trust_best_applicants  ...   acceptance_wages_package  \
0           Très faible  ...  Ni réticent, ni favorable   
1   Ni faible, ni élevé  ...              Très réticent   
2           Très faible  ...                   Réticent   
3                Faible  ...              Très réticent   
4   Ni faible, ni élevé  ...                   Réticent   
5                Faible  ...                   Réticent   
6   Ni faible, ni élevé  ...              Très réticent   
7                Faible  ...              Très réticent   
8           Très faible  ...              Très réticent   
9                Faible  ...              Très réticent   

         acceptance_contract          ethics_privacy  \
0                  Favorable  C'est plutôt important   
1                   Réticent  C'est plutôt important   
2                  Favorable    C'est très important   
3  Ni réticent, ni favorable  C'est plutôt important   
4  Ni réticent, ni favorable   C'est très secondaire   
5  Ni réticent, ni favorable    C'est très important   
6              Très réticent  C'est plutôt important   
7              Très réticent    C'est très important   
8              Très réticent    C'est très important   
9                  Favorable    C'est très imp

## Remplacement dans les réponses des valeurs alphanumériques par des valeurs numériques

**L'échelle de Likert à 5 valeurs est utilisée. La valeur neutre est toujours la 3ème** <br>
**La valeur ayant la connotattion la plus négative et/ou secondaire est égale à 1, la valeur la plus positive et/ou importante est égale à 5**


In [143]:
likert_mapping_usefulness = {"Totalement opposé":1,
                    "Plutôt opposé":2,
                    "Ni favorable, ni opposé":3,
                    "Assez favorable":4,
                    "Totalement favorable":5}

likert_mapping_easiness = {"à l\'aise":5,
                    "Plutôt à l\'aise":4,
                    "Ni aisance, ni difficultés":3,
                    "Quelques difficultés":2,
                    "En difficulté":1}

likert_mapping_trust = {"Très faible":1,
                        "Faible":2,
                        "Ni faible, ni élevé":3,
                        "Elevé":4,
                        "Très élevé":5}

likert_mapping_acceptance = {"Très réticent":1,
                            "Réticent":2,
                            "Ni réticent, ni favorable":3,
                            "Favorable":4,
                            "Très favorable":5}

likert_mapping_ethics = {"C\'est très important":5,
                        "C\'est plutôt important":4,
                        "C\'est important ou secondaire":3,
                        "C\'est plutôt secondaire":2,
                        "C\'est très secondaire":1}

likert_mapping_AI_hiring = {"C\'est essentiel":5,
                            "C\'est important":4,
                            "Cela dépend":3,
                            "C\'est secondaire":2,
                            "C\'est sans intérêt":1}


# Chiffres essentiels

## Nombre de réponses

In [68]:
shape = df.shape

In [69]:
print(f"Le questionnaire comporte {shape[0]} réponses avec chacune {shape[1]} items")

Le questionnaire comporte 161 réponses avec chacune 27 items


## Répartition de l'âge des répondants

In [171]:
df.count()

useful_resume_CV                  161
useful_linkedin                   161
useful_interview                  161
useful_intelligence               161
useful_speedy_response            161
easiness_generative_AI            161
easiness_search_engine            161
easiness_navigation               161
easiness_business_sciences        161
trust_best_applicants             161
trust_discrimination              161
trust_ethics                      161
trust_diversity                   161
acceptance_sorting_resumes        161
acceptance_selection_tests        161
acceptance_interviews             161
acceptance_negotiation            161
acceptance_wages_package          161
acceptance_contract               161
ethics_privacy                    161
ethics_carbon_footprint           159
ethics_no_discrimination          161
ethics_fairness_for_applicants    161
ethics_diversity                  159
misc_recruiter_uses_AI            161
misc_age_category                 161
misc_free_te

### Commentaire en première approche sur le nombre de réponses.

- 161 personnes ont répondu. La totalité des questions a obtenu 161 réponses
- 3 questions ont obtenu un moins de 161 réponses : celle portant sur l'empreinte carbone (159 réponses), sur la diversité (159 réponses) et celle du commentaire libre (67 réponses).
- Répondre sur l'éthique de l'IA comme le commentaire libre était optionnel, mais indiqué comme tel uniquement pour le commentaire libre.

### Pourcentage de réponses pour les questions optionnelles

In [206]:
# Pourcentage de réponses pour les questions optionnelles
counts = dict(df.count())
ethics_diversity = int(counts.get("ethics_diversity"))
ethics_carbon_footprint = int(counts.get("ethics_carbon_footprint"))
free_text = int(counts.get("misc_free_text"))

percent_diversity = round(ethics_diversity/shape[0]*100, 1)
percent_carbon_footprint = round(ethics_carbon_footprint/shape[0]*100, 1)
free_text= round(free_text/shape[0]*100, 1)

In [208]:
print(f"{percent_diversity}% des répondants se sont positionnés sur l'importance accordée à la diversité.")
print(f"{percent_diversity}% des répondants se sont positionnés sur l'importance accordée à l'empreinte carbone.")
print(f"{free_text}% des répondants ont laissé un commentaire en texte libre à la fin du questionnaire.")

98.8% des répondants se sont positionnés sur l'importance accordée à la diversité.
98.8% des répondants se sont positionnés sur l'importance accordée à l'empreinte carbone.
41.6% des répondants ont laissé un commentaire en texte libre à la fin du questionnaire.


# Calcul de la fiabilité du questionnaire

**Nous allons calculer l'intervalle de confiance et la marge d'erreur sur la base du nombre de réponses au questionnaire**

In [72]:
import math

In [88]:
# Usually confidence level is either 0.90, 0.95 or 0.99, depending on the expected strength of the questionnaire
confidence_level = 0.95

# The estimated ratio p defines the maximum margin of error. 0.5 is a standard value in case of lack of prior reference
p = 0.5

## Calcul de la valeur de score z

In [89]:
import scipy.stats

In [92]:
def calculate_z(confidence_level):
    """
    Calculate z score depending on chosen confidence level.

    Arguments:
        confidence_level: represents the level of confidence expressing the confidence that the sample represents the overall population. 
        Usual values for z are : 0.9, 0.95 or 0.99. The higher the figure, the stronger the level of confidence.

    Returns:
        z score: used to subsequently calculate a margin of error depending on confidence level.
    """
    
    cumulative_probability = (1 + confidence_level)/2
    z = scipy.stats.norm.ppf(cumulative_probability)

    return z

In [93]:
z_score = calculate_z(confidence_level)

In [96]:
print(f"La valeur du score z pour un niveau de confiance de {confidence_level} est de {round(z_score, 2)}")

La valeur du score z pour un niveau de confiance de 0.95 est de 1.96


## Correction en fonction de la population active

In [74]:
# total french workforce
total_workforce = 30000000

# sample size
sample_size = shape[0]

In [84]:
def population_correction(total_population, sample_size):
    """ Calculate the finished population correction factor due to sample size.

        Parameters:
        total_population: total population to be considered.
        sample_size: number of responses in the questionnaire.

        Returns:
        pop_correction_factor: factor determining the optimal sample size considering the overall finished population. 
        A 1.0 factor means the sample size is not ideal with regards to the population.
    """
    pop_correction_factor = math.sqrt((total_population - sample_size) / (total_workforce - 1))
    return pop_correction_factor

In [85]:
pop_correction_factor = population_correction(total_workforce, sample_size)

In [83]:
print(f"Le facteur de correction de la population finie est de {round(pop_correction_factor, 0)}")

Le facteur de correction de la population finie est de 1.0


## Calcul de l'intervalle de confiance

In [98]:
def confidence_interval(confidence_level, total_population, sample_size, z_score, pop_correction_factor,  p=0.5):
    """
    Calculate the confidence interval and the margin of error of a defined sample.

    Params:
        confidence_level: represents the level of confidence expressing the confidence that the sample represents the overall population. 
        Usual values for z are : 0.9, 0.95 or 0.99. The higher the figure, the stronger the level of confidence.
        p: represents the proportion of the population presenting a given characteristics. Value by default is 0.5.
        total_population: size of the population considered for the survey.
        sample_size: number of persons having participated to the survey and whose answers can be used.
        z_score: deviation between a given value and the mean of a dataset.
        pop_correction_factor: helps to define optimal sample size

    Returns : 
        error_margins: a tuple containing the upper, lower margins of error and redressed_error_margin
    """

    error_margin = z_score * math.sqrt((p * (1 - p)) / sample_size)

    redressed_error_margin = error_margin * pop_correction_factor

    lower_margin = p - redressed_error_margin
    upper_margin = p + redressed_error_margin

    return (upper_margin, lower_margin, redressed_error_margin)

In [116]:
upper_margin, lower_margin, error_margin = confidence_interval(confidence_level, total_workforce, sample_size, z_score, pop_correction_factor, p)

In [123]:
print(f"Pour un niveau de confiance de {confidence_level*100}% et une population active de {total_workforce}, avec un échantillon de {sample_size} répondants et une valeur p = {p}:")
print(f"La valeur inférieure est : {round(lower_margin *100, 2)}%")
print(f"La valeur supérieure est : {round(upper_margin *100, 2)}%")
print(f"La marge d'erreur est : {round(error_margin *100, 2)}%")

Pour un niveau de confiance de 95.0% et une population active de 30000000, avec un échantillon de 161 répondants et une valeur p = 0.5:
La valeur inférieure est : 42.28%
La valeur supérieure est : 57.72%
La marge d'erreur est : 7.72%
